frontends/web/src/routes/buy/exchange.tsx

```diff
@@ -105,46 +108,19 @@ export const Exchange = ({ code, accounts }: TProps) => {
 
   }, [isPocketSupportedCoin, isMoonpaySupportedCoin, selectedRegion, regionList]);
 
-  const findBestDeal = (deals: exchangesAPI.ExchangeDeal[]): exchangesAPI.ExchangeDeal => {
-    let best = deals[0];
-    deals.forEach(deal => best = deal.fee < best.fee ? deal : best);
-    return best;
-  };
 
-  // set the bestDeal when getExchangeDeals returns.
+  // re-structure and store all exchange
+  // providers into a state to be rendered
   useEffect(() => {
     if (!exchangeDeals) {
       return;
     }
-    const pocketBest = findBestDeal(exchangeDeals.pocket);
-    const moonpayBest = findBestDeal(exchangeDeals.moonpay);
-    setBestDeal(findBestDeal([pocketBest, moonpayBest]));
-  }, [exchangeDeals]);
 
-  const dealsDetails = (exchange: string): string => {
-    if (!exchangeDeals) {
-      return '';
-    }
-    var details = exchange === 'pocket' ? 'Pocket' : 'Moonpay';
-    const deals = exchange === 'pocket' ? exchangeDeals.pocket : exchangeDeals.moonpay;
-    deals.forEach(deal => {
-      details += ' | ';
-      details += deal.payment === 'card' ? t('buy.exchange.creditCard') : t('buy.exchange.bankTransfer');
-      details += ' - ' + t('buy.exchange.fee') + ': ';
-      details += String(deal.fee * 100) + '%';
-      if (deal === bestDeal && showPocket && showMoonpay) {
-        details += ' - ';
-        //TODO replace with icon
-        details += 'BEST DEAL';
-      }
-      if (deal.isFast) {
-        details += ' - ';
-        //TODO replace with icon
-        details += 'FAST';
-      }
-    });
-    return details;
-  };
+    const exchanges = findBestDeal(exchangeDeals).map(ex => ({ ...ex, supported: ex.providerName === 'pocket' ? isPocketSupportedCoin : isMoonpaySupportedCoin }));
```

Comment:
I guess you should use `showPocket` (which depends on the account and the selected region) instead of `isPocketSupportedCoin` (which depends only on account). Otherwise, you won't filter correctly among regions. It seems to me that the current behavior shows both exchanges when at least one of theme is available. E.g. Pocket is available in Algeria, while it shouldn't (see https://widget.pocketbitcoin.com/api/availabilities for pocket availabilities)

Reply:
Also, I think that the `best deal` logic should depend on the selected country. Otherwise, if the provider with the lowest fee is not available in a specific region we won't have a `best deal` option in that region (see the note in my other comment about that :) )

Reply:
Thanks for this! Will use `showPocket` and `showMoonPay` then.